doc/aaa/radius_authentication.md

```diff
@@ -0,0 +1,583 @@
+|         |     PAM     |        |       +----------^----------+
+|         |  Libraries  |        |                  |
+|         +-------------+        |              (2) |
++--------------------------------+                  |
+                                                    |
+                                                    |
+           +---------+                      +-------+--------+
+           |         |        (1)           |                |
+           |   CLI   +---------------------->    ConfigDB    |
+           |         |                      |                |
+           +---------+                      +----------------+
+
+```
+
+
+Following is the sequence of events during configuring RADIUS authentication:
+
+1. The CLI is used to update the ConfigDB with the RADIUS and AAA configuration.
+2. The Host Config Daemon(HCD) reads data from ConfigDB.
+3. The HCD configures the PAM authentication files. (The AAA config module in
+   the HCD is responsible for modifying PAM configuration files.)
+4. All new CLI sessions (SSH, Console), now use the new PAM configurations
+   to authenticate users.
+
+Following is the sequence of events during RADIUS authenticated login to
+the Console:
+
+```
+Console i/p           SONiC Device                     RADIUS Server
+                  /bin/login     NSS RADIUS Plugin
+----------------------------------------------------------------------
+
+    |                 |                 |                    |
+    |                 |                 |                    |
+    |  Login(1)       |                 |                    |
+    +---------------->|                                      |
+    |                 |           Access-Request(2)          |
+    |                 +------------------------------------->|
+    |                 |                                      |
+    |                 |           Access-Accept(3)           |
+    |                 |<-------------------------------------+
+    |                 |                                      |
+    |             (4) |                 |                    |
+    |                 | getpwnam_r()(5) |                    |
+    |                 +---------------->|                    |
+    |                 |                 |                    |
+    |                 | user info.(6)   |                    |
+    |                 |<----------------+                    |
+    |                 |                 |                    |
+    |  Success(7)     |                 |                    |
+    |<----------------|                 |                    |
+    |                 |                 |                    |
+
+```
+
+1. A user tries to login through the Console
+2. The PAM configuration causes an Access-Request to RADIUS server.
+3. RADIUS returns Access-Accept with Management-Privilege-Level (MPL)(Goto 4.),
+   or Access-Reject (Authentication Failed).
```

Comment:
Where does the Cached MPL (management-privilege-level)stored ? 
Can the MPL cache associate with TTL?
How about user login's on device reboots? does it expect login failed/success?

Reply:
> Where does the Cached MPL (management-privilege-level)stored ? 

Cached MPL is stored under /var/run/radius. This is mentioned on line 271

> Can the MPL cache associate with TTL?

Yes, every time the user logs in, their cache MPL is updated.

> How about user login's on device reboots? does it expect login failed/success?

I didn't get the question here. Is the inquiry regarding cached MPL cached across device reboot. The cached contents are in /var/run/radius, and are updated when the user logs in.

Reply:
Updated the the location of the cache to /var/cache/radius. since that is recommended place for Application cache data